src/conv-core/convcore.C

```diff
@@ -3093,6 +3093,7 @@ void CmiInitMsgHeader(void *msg, int size) {
   if(size >= CmiMsgHeaderSizeBytes)
     CMI_ZC_MSGTYPE(msg) = CMK_REG_NO_ZC_MSG;
 #endif
+  CMI_MSG_NOKEEP(msg) = 0;
```

Comment:
`CMI_MSG_NOKEEP(msg) = 0;` should be guarded with an `if(size >= CmiMsgHeaderSizeBytes)` because users (or even within RTS) there can be smaller allocations using CmiAlloc. 

I think it's best to do the following:

```
  if(size >= CmiMsgHeaderSizeBytes)  {
#if CMK_ONESIDED_IMPL
    // Set zcMsgType in the converse message header to CMK_REG_NO_ZC_MSG
    CMI_ZC_MSGTYPE(msg) = CMK_REG_NO_ZC_MSG;
#endif
    CMI_MSG_NOKEEP(msg) = 0;
  }
```